frontend/src/views/Collection/utils.ts

```diff
@@ -265,6 +265,24 @@ export function isCollectionHasPrivateRevision(
   );
 }
 
+/**
+ * Returns sorted datasets based on the given dataset order.
+ * @param datasets - Datasets.
+ * @param orderedIds - Dataset IDs, ordered.
+ * @returns sorted datasets.
+ */
+export function sortCollectionDatasets(
+  datasets: Dataset[],
+  orderedIds?: string[]
+): Dataset[] {
+  // Early exit; return datasets - reordering mode is inactive.
+  if (!orderedIds) return datasets;
+  // Return datasets in order - reordering mode is active.
+  return orderedIds
```

Comment:
is it worth caching the computed result to prevent creating a new array when the arguments are the same 😄 ?

Reply:
Thanks for taking a look at the PR @tihuan! 🌠 
WRT updating the datasets, `sortCollectionDatasets` is called when the `orderedIds` is updated on drop and only when the `orderedIds` have changed i.e. if the user drops back onto the original row the `Collection` does not experience a re-render. Let me know if this is what you were hoping for?

Reply:
Sorry for the late reply, @frano-m ! It was CZI offsite last week 😆 

Appreciate your quick response too 🙌 ! 

I see that if `orderedIds` is absent, then the original `datasets` is returned 👍 

But I was just wondering if there's a case where `sortCollectionDatasets` could be called with the same `datasets` and `orderedIds` twice in a row that we could avoid a re-render by returning a cached re-ordered `datasets`? If that's not possible then please ignore my comment 😂 

Thanks again!